packages/core/src/hydration/api.ts

```diff
@@ -162,6 +162,24 @@ export function withDomHydration(): EnvironmentProviders {
         // be replaced with a tree-shakable alternative (e.g. `isServer`
         // flag).
         if (isBrowser() && inject(IS_HYDRATION_DOM_REUSE_ENABLED)) {
+          // On the client, the DOM must contain a hydration marker or hydration is broken
+          const doc = getDocument();
+          let hydrationMarker: Node|undefined;
+          for (const node of doc.body.childNodes) {
+            if (node.nodeType === Node.COMMENT_NODE &&
+                node.textContent === SSR_CONTENT_INTEGRITY_MARKER) {
```

Comment:
I'd propose doing trim here just in case:
```suggestion
                node.textContent.trim() === SSR_CONTENT_INTEGRITY_MARKER) {
```